neuron_morphology/pipeline/post_data_to_s3.py

```diff
@@ -0,0 +1,134 @@
+import os
+import boto3
+import json
+import configparser
+from zipfile import ZipFile, ZIP_DEFLATED
+from io import BytesIO
+from argschema import ArgSchemaParser
+from neuron_morphology.pipeline._schemas import InputParameters
+
+def get_credentials(credentials_file):
+    """
+    get credentisals from credentials_file
+
+    Parameters
+    ---------------
+    credentials_file: file path to credentials_file
+
+    Return
+    ---------------
+    credentials: access_key_id, secret_access_key
+
+    """
+    config = configparser.ConfigParser()
+    config.read(credentials_file)
+    access_key_id = None
+    secret_access_key = None
+    if 'default' in config.sections():
+        if 'aws_access_key_id' in config.options('default'):
+            access_key_id = config.get('default', 'aws_access_key_id')
+        if 'aws_access_key_id' in config.options('default'):
+            secret_access_key = config.get('default', 'aws_secret_access_key')
+
+    return access_key_id, secret_access_key
+
+def zip_files(file_dict):
+    """
+    zip files into an archive in memory
+
+    Parameters
+    ---------------
+    file_dict: file name: file paths or file in bytes to be archived
+
+    Return
+    ---------------
+    archive: data in bytes
+
+    """
+
+    archive = BytesIO()
+
+    with ZipFile(archive, "a", ZIP_DEFLATED, False) as zip_file:
+        for name, file in file_dict.items():
+            if isinstance(file, str):
+                with open(file, 'rb') as fh:
+                    data = BytesIO(fh.read())
+                zip_file.writestr(name, data.getvalue())
+            elif isinstance(file, BytesIO):
+                zip_file.writestr(name, file.getvalue())
+            else:
+                raise TypeError("Invalid input file!")
+
+    return archive.getvalue()
+
+
+def post_object_to_s3(data, name, bucket, region, access_key_id=None, secret_access_key=None):
+    """
+    This zip files to an archive in memory and post it to S3 bucket
```

Comment:
I have noticed that we still have a mention of a zip file/archive in here. You may also document here where credentials are coming from in not explicitly provided.

Reply:
If boto3 fails in finding /allen/aibs/technology/production/.aws/credentials you can always set AWS_SHARED_CREDENTIALS_FILE=/allen/aibs/technology/production/.aws/credentials in your env.

revised the doc as suggested.